src/main/scala/org/broadinstitute/dsde/rawls/user/UserService.scala

```diff
@@ -415,43 +418,71 @@ class UserService(protected val userInfo: UserInfo, val dataSource: SlickDataSou
         DBIO.failed(new RawlsExceptionWithErrorReport(errorReport = ErrorReport(StatusCodes.Conflict, s"Cannot create billing project [${projectName.value}] in database because it already exists")))
       case None =>
-        createBillingProjectInternal(dataAccess, projectName, Set.empty, Set.empty, CreationStatuses.Ready) map (_ => RequestComplete(StatusCodes.Created))
+        for {
+          groups <- createBillingProjectGroups(dataAccess, projectName)
+          project <- createBillingProjectInternal(dataAccess, projectName, CreationStatuses.Ready, groups) map (_ => RequestComplete(StatusCodes.Created))
+        } yield project
     }
   }
 
-  private def createBillingProjectInternal(dataAccess: DataAccess, projectName: RawlsBillingProjectName, owners: Set[RawlsUserRef], users: Set[RawlsUserRef], status: CreationStatuses.CreationStatus): WriteAction[RawlsBillingProject] = {
+  private def createBillingProjectGroups(dataAccess: DataAccess, projectName: RawlsBillingProjectName, creators: Set[RawlsUserRef] = Set.empty): ReadWriteAction[Map[ProjectRoles.ProjectRole, RawlsGroup]] = {
+    val createGroups = DBIO.sequence(ProjectRoles.all.map { role =>
+      val name = RawlsGroupName(gcsDAO.toBillingProjectGroupName(projectName, role))
+      createGroupInternal(RawlsGroupRef(name), dataAccess).map(g => role -> g)
+    }.toSeq)
+
+    createGroups flatMap { groups =>
+      val groupMap = groups.toMap
+      DBIO.sequence(creators.map(dataAccess.rawlsUserQuery.load).toSeq) flatMap { owners =>
+        updateGroupMembersInternal(groupMap(ProjectRoles.Owner), owners.flatten.toSet, Set.empty, AddGroupMembersOp, dataAccess).map(_ => groupMap)
+      }
+    }
+  }
+
+  private def createBillingProjectInternal(dataAccess: DataAccess, projectName: RawlsBillingProjectName, status: CreationStatuses.CreationStatus, groups: Map[ProjectRoles.ProjectRole, RawlsGroup] = Map.empty): ReadWriteAction[RawlsBillingProject] = {
     DBIO.from(gcsDAO.createCromwellAuthBucket(projectName)) flatMap { bucketName =>
       val bucketUrl = "gs://" + bucketName
-      dataAccess.rawlsBillingProjectQuery.save(RawlsBillingProject(projectName, owners, users, bucketUrl, status))
+
+      val billingProject = RawlsBillingProject(projectName, groups, bucketUrl, status)
+      dataAccess.rawlsBillingProjectQuery.create(billingProject)
     }
   }
 
   def unregisterBillingProject(projectName: RawlsBillingProjectName): Future[PerRequestMessage] = dataSource.inTransaction { dataAccess =>
     withBillingProject(projectName, dataAccess) { project =>
-      dataAccess.rawlsBillingProjectQuery.delete(project) map {
-        case true => RequestComplete(StatusCodes.OK)
-        case false => throw new RawlsExceptionWithErrorReport(errorReport = ErrorReport(StatusCodes.InternalServerError, s"Could not delete billing project [${projectName.value}]"))
-      }
+      DBIO.from(Future.sequence(project.groups.map {case (_, group) => gcsDAO.deleteGoogleGroup(group) })) andThen
```

Comment:
Google call inside transaction.  Can we make `unregisterBillingProject` two transactions, like `WorkspaceService.copyMethodConfiguration` ?